api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -77,6 +78,12 @@ public void validate(Object obj, Errors errors) {
 		} else if (obs.getVoided()) {
 			return;
 		}
+		if (obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null) {
+			if (!Objects.equals(obs.getPerson(),obs.getEncounter().getPatient().getPerson())) {
+			  errors.reject("Person objects of the obs and the encounter do not match");
+			}
+		}
+		
```

Comment:
Shouldn't this be down in the `validateHelper()` method? That way we also run the validation for child obs.

Reply:
ok @ibacher ,let me add it at that point